pensieve/config.py

```diff
@@ -26,30 +26,90 @@
+from . import AnalysisPeriod
+from pensieve.statistics import Statistic, BootstrapMean, StatisticResultCollection
 import pensieve.experimenter
 
 
-# TODO: these should be objects that pair a metric with a default statistical treatment.
+@attr.s(auto_attribs=True)
+class Summary:
+    """Represents a metric with a statistical treatment."""
+
+    metric: mozanalysis.metrics.Metric
+    treatment: Statistic
+
+    def run(self, data: pandas.DataFrame) -> "StatisticResultCollection":
+        """Apply the statistic transformation for data related to the specified metric."""
+        return self.treatment.apply(data, self.metric.name)
+
+
+# metrics and statistics calculated for each experiment
 DEFAULT_METRICS = {
     "desktop": {
-        "daily": [mozanalysis.metrics.desktop.unenroll],
-        "weekly": [
-            mozanalysis.metrics.desktop.active_hours,
-            mozanalysis.metrics.desktop.uri_count,
-            mozanalysis.metrics.desktop.ad_clicks,
-            mozanalysis.metrics.desktop.search_count,
+        AnalysisPeriod.DAY: [
+            Summary(
+                metric=mozanalysis.metrics.desktop.unenroll,
+                treatment=BootstrapMean(num_samples=1000),
+            )
         ],
-        "overall": [
-            mozanalysis.metrics.desktop.active_hours,
-            mozanalysis.metrics.desktop.uri_count,
-            mozanalysis.metrics.desktop.ad_clicks,
-            mozanalysis.metrics.desktop.search_count,
+        AnalysisPeriod.WEEK: [
+            Summary(
+                metric=mozanalysis.metrics.desktop.active_hours,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.uri_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.ad_clicks,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.search_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+        ],
+        AnalysisPeriod.OVERALL: [
+            Summary(
+                metric=mozanalysis.metrics.desktop.active_hours,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.uri_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.ad_clicks,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
+            Summary(
+                metric=mozanalysis.metrics.desktop.search_count,
+                treatment=BootstrapMean(num_samples=1000),
+            ),
         ],
     }
 }
 
+# metrics that are available in mozanalysis with a default statistical treatment
+AVAILABLE_METRICS = {
```

Comment:
I'm not entirely sure what the best way is to define default statistics for existing metrics. It seems to me that there is no one-fits-all way for defining a default treatment. 

The best I could come up with for now is defining default statistics explicitly for existing metrics. This would require creating PRs every time a new metric gets added to mozanalysis, however currently we also need to update the mozanalysis version for this. So we'd need to update pensieve in any case. This is just a lot more verbose, which might not be a bad thing?

Reply:
> It seems to me that there is no one-fits-all way for defining a default treatment.

How do you mean? I don't really think we need to try to get it right. This approach makes sense to me, though.